### PR TITLE
Prevent primary notification action from displaying in secondary actions list

### PR DESCRIPTION
**Pull request by staykids**
**Phabricator:** https://phabricator.wikimedia.org/T302453 (partially)

### Notes
Adds a shallow URL check of the primary action and secondary actions to prevent the secondary actions from displaying any action that has the same destination as the primary action.

### Test Steps
Perform a light regression test of the notifications detail view to confirm nothing unexpected happens when displaying or tapping secondary actions. I don't have a particular type where I've personally noticed the duplication to begin with, so perhaps we should rely on QA to test through the available remote notification types.

### CODE REVIEW

**Review by tonisevener — CHANGES_REQUESTED** [review]
Thanks @staykids! I found a few more cases where I think we need to add some deduping (see comments). It does seem like user talk page edits are the ones that have the dupes - I'd still like to keep this as a sort of post-processing thing (like you have here) rather than explicitly check for that namespace at a deeper level, just in case it's happening on other namespace types too.

If it helps, I constructed a payload of my problem notifications that you could try mapping locally against for testing:

[problem-notifications.txt](https://github.com/wikimedia/wikipedia-ios/files/8535292/problem-notifications.txt)

File: Wikipedia/Code/NotificationsCenterDetailViewModel+ActionExtensions.swift [hunk 1]
```diff
@@ -124,6 +124,18 @@ extension NotificationsCenterDetailViewModel {
...
+    /// Do not include secondary action if its destination is the same as the primary action
+    var uniqueSecondaryActions: [NotificationsCenterAction] {
+        guard let primaryActionURL = primaryAction?.actionData?.url else {
+            return secondaryActions
+        }
+
+        return secondaryActions.filter({ action in
+            action.actionData?.url != primaryActionURL
+        })
+    }
```
[the remark this answers is not in the document]
Comment (staykids) [reply Wikipedia/Code/NotificationsCenterDetailViewModel+ActionExtensions.swift:139]: Interesting! I didn't consider duplicates within the secondary actions themselves.

Keeping this at the post-processing level, I just added `Hashable` conformance to `NotificationsCenterIconType` to be able to use Set to remove duplicates from both the secondary actions and the sheet actions. Does this solution feel sufficient to you?

I think it's ready for another look!

[the remark this answers is not in the document]
Comment (tonisevener) [reply Wikipedia/Code/NotificationsCenterDetailViewModel+ActionExtensions.swift:139]: @staykids thanks! This approach sounds fine to me - though the deduping didn't work for me until I also conformed `NotificationsCenterAction` and `NotificationsCenterActionData` to `Hashable`. So just that one last change is needed I think.

[the remark this answers is not in the document]
Comment (staykids) [reply Wikipedia/Code/NotificationsCenterDetailViewModel+ActionExtensions.swift:139]: Strange! I must be misunderstanding or missing something, because I expected that all the nested `struct`'s properties conforming to `Equatable` would be sufficient.

Updated! Also updated the tests to reflect the expected sheet actions after removing duplicates.

[the remark this answers is not in the document]
Comment (tonisevener) [reply Wikipedia/Code/NotificationsCenterDetailViewModel+ActionExtensions.swift:139]: No problem. I'm actually a little surprised it worked with `NSOrderedSet` at all, I thought you'd have to use a more Objective-C friendly object 😅. Seems to be working well though!
